lib/javascripts/base_app.js

```diff
@@ -46,17 +46,17 @@ function registerHelpers(app) {
 
 function BaseApp(zafClient, data) {
   this.zafClient = zafClient;
-  this.I18n = { t: I18n.t.bind(null, this) };
+  this.I18n = { t: I18n.t };
   registerHelpers(this);
   bindEvents(this);
   let evt = { firstLoad: true };
   this._metadata = data.metadata;
   this._context = data.context;
-  if (this.defaultState) {
-    this.switchTo(this.defaultState);
-  }
   zafClient.get('currentUser.locale').then(function(data) {
-    I18n.loadTranslations(this, data['currentUser.locale']);
+    I18n.loadTranslations(data['currentUser.locale']);
+    if (this.defaultState) {
```

Comment:
@joseconsador this is the actual fix - move the `switchTo(this.defaultState)` after `loadTranslations`